File: src/dataset/typescript/acronym_exercism-wackerow/claude/ClaudeSonnet4_acronym_v3.ts
export default class Acronym {
  public static parse(phrase: string): string {
    const result: string[] = [];
    let takeNext = true;
    
    for (let i = 0; i < phrase.length; i++) {
      const char = phrase[i];
      
      if (char === ':') break;
      
      if (char === ' ' || char === '-') {
        takeNext = true;
      } else if (takeNext && char >= 'A' && char <= 'z') {
        result.push(char.toUpperCase());
        takeNext = false;
      } else if (char >= 'A' && char <= 'Z') {
        result.push(char);
        takeNext = false;
      }
    }
    
    return result.join('');
  }
}

File: src/dataset/Java/acronym_exercism-awsaf174/claude/ClaudeSonnet4_Acronym_v3.java
class Acronym {

  private final String acronym;

  Acronym(String phrase) {
    if (phrase == null || phrase.isEmpty()) {
      acronym = "";
      return;
    }

    StringBuilder result = new StringBuilder();
    boolean nextIsAcronym = true;
    
    for (int i = 0; i < phrase.length(); i++) {
      char c = phrase.charAt(i);
      
      if (Character.isLetter(c) || Character.isDigit(c)) {
        if (nextIsAcronym) {
          result.append(Character.toUpperCase(c));
          nextIsAcronym = false;
        }
      } else {
        nextIsAcronym = true;
      }
    }
    
    acronym = result.toString();
  }

  String get() {
    return acronym;
  }
}